web/frontend/src/pages/form/IndividualResult.tsx

```diff
@@ -62,82 +62,89 @@ const IndividualResult: FC<IndividualResultProps> = ({
+              {i18n.language === 'de' && titles.de}
+            </h2>
           </div>
-          <h2 className="flex align-text-middle text-lg pb-2">
-            {i18n.language === 'en' && titles.en}
-            {i18n.language === 'fr' && titles.fr}
-            {i18n.language === 'de' && titles.de}
-          </h2>
+          {element.Type === RANK && rankResult.has(element.ID) && (
+            <IndividualRankResult
+              rank={element as RankQuestion}
+              rankResult={[rankResult.get(element.ID)[internalID]]}
+            />
+          )}
+          {element.Type === SELECT && selectResult.has(element.ID) && (
+            <IndividualSelectResult
+              select={element as SelectQuestion}
+              selectResult={[selectResult.get(element.ID)[internalID]]}
+            />
+          )}
+          {element.Type === TEXT && textResult.has(element.ID) && (
+            <IndividualTextResult
+              text={element as TextQuestion}
+              textResult={[textResult.get(element.ID)[internalID]]}
+            />
+          )}
         </div>
-        {element.Type === RANK && rankResult.has(element.ID) && (
-          <IndividualRankResult
-            rank={element as RankQuestion}
-            rankResult={[rankResult.get(element.ID)[internalID]]}
-          />
-        )}
-        {element.Type === SELECT && selectResult.has(element.ID) && (
-          <IndividualSelectResult
-            select={element as SelectQuestion}
-            selectResult={[selectResult.get(element.ID)[internalID]]}
-          />
-        )}
-        {element.Type === TEXT && textResult.has(element.ID) && (
-          <IndividualTextResult
-            text={element as TextQuestion}
-            textResult={[textResult.get(element.ID)[internalID]]}
-          />
-        )}
-      </div>
-    );
-  };
+      );
+    },
+    [internalID, rankResult, selectResult, textResult]
+  );
 
-  const displayResults = (subject: Subject) => {
-    let sbj;
-    if (isJson(subject.Title)) {
-      sbj = JSON.parse(subject.Title);
-    }
-    if (sbj === undefined) {
-      sbj = { en: subject.Title, fr: subject.TitleFr, de: subject.TitleDe };
-    }
-    return (
-      <div key={subject.ID}>
-        <h2 className="text-xl pt-1 pb-1 sm:pt-2 sm:pb-2 border-t font-bold text-gray-600">
-          {i18n.language === 'en' && sbj.en}
-          {i18n.language === 'fr' && sbj.fr}
-          {i18n.language === 'de' && sbj.de}
-        </h2>
-        {subject.Order.map((id: ID) => (
-          <div key={id}>
-            {subject.Elements.get(id).Type === SUBJECT ? (
-              <div className="pl-4 sm:pl-6">
-                {displayResults(subject.Elements.get(id) as Subject)}
-              </div>
-            ) : (
-              SubjectElementResultDisplay(subject.Elements.get(id))
-            )}
-          </div>
-        ))}
-      </div>
-    );
-  };
+  const displayResults = useCallback(
```

Comment:
See https://dev.to/amitaldo/function-makes-the-dependencies-of-useeffect-hook-change-on-every-render-warning-in-react-35hd

Reply:
same, this looks like it should be moved to its own component, maybe combined with the one above.